src/KustoExpressionParser.ts

```diff
@@ -43,17 +43,30 @@ export class KustoExpressionParser {
     };
 
     const parts: string[] = [];
+    const expandParts: string[] = [];
+    const isDynamicArray = query.search.property.name.includes(DYNAMIC_TYPE_ARRAY_DELIMITER);
+    const name: string = this.addExpanPartsdIfNeeded(query.search.property.name, expandParts);
+    const column = context.castIfDynamic(name, query.search.property.name);
     this.appendProperty(context, query.expression.from, parts);
     this.appendTimeFilter(context, undefined, parts, tableSchema);
+    this.appendMvExpand(expandParts, parts);
+
+    if (expandParts.length) {
```

Comment:
nit: this can be `if (isDynamicArray)` (you may want to move it under the other else?